e2e/tests/update_params_test.go

```diff
@@ -0,0 +1,341 @@
+		case "module":
+			*addr = authtypes.NewModuleAddress(name).String()
+		case "user":
+			*addr = s.getKeyAddress(name)
+		default:
+			s.Fatal("unexpected address type")
+		}
+	}
+
+	// Query the authz module for grants with the expected granter & grantee (authz.QueryGrantsRequest).
+	argsAndFlags := []string{
+		"query", "authz", "grants",
+		granterAddr, granteeAddr, msgType,
+		fmt.Sprintf("--%s=json", cometcli.OutputFlag),
+	}
+	res, err := s.pocketd.RunCommandOnHost("", argsAndFlags...)
+	require.NoError(s, err)
+
+	// Unmarshal the JSON response into the grantsRes struct.
+	// Using s.cdc.UnmarshalJSON() with *authz.QueryGrantsResponse complains about
+	// the "type" field, expecting "@type" for the #Grants[n]#Authorization pb.Any.
+	grantsRes := new(authzCLIGrantResponse)
+	responseBz := []byte(strings.TrimSpace(res.Stdout))
+	err = json.Unmarshal(responseBz, grantsRes)
+	require.NoError(s, err)
+
+	// Check if the expected grant is found in the response.
+	grantFound := false
+	for _, grant := range grantsRes.Grants {
+		if grant.Authorization.Value.Msg == msgType {
+			grantFound = true
+			break
+		}
+	}
+	require.True(s, grantFound)
+
+	// Update the state of the suite with the granter & grantee names.
+	s.granterName = granterName
+	s.granteeName = granteeName
+}
+
+func (s *suite) TheUserSendsAnAuthzExecMessageToUpdateAllModuleParams(moduleName string, table gocuke.DataTable) {
+	ctx := context.Background()
+
+	// Convert the text table to a map of param names to their types & values.
+	paramsMap := s.parseParamsTable(table)
+
+	// Use the map of params to populate a tx JSON template & write it to a file.
+	txJSONPath := s.newTempTxJSONFile(moduleName, paramsMap)
+
+	argsAndFlags := []string{
+		"tx", "authz", "exec",
+		txJSONPath,
+		"--from", s.granteeName,
+		keyRingFlag,
+		fmt.Sprintf("--%s=json", cometcli.OutputFlag),
+	}
+	_, err := s.pocketd.RunCommandOnHost("", argsAndFlags...)
+	require.NoError(s, err)
+
+	// Reset all module params to their default values after the test completes.
+	s.Cleanup(func() { s.resetAllModuleParamsToDefaults(ctx) })
+}
+
+func (s *suite) AllModuleParamsShouldBeUpdated(moduleName string, table gocuke.DataTable) {
+	panic("PENDING")
+}
+
+func (s *suite) TheUserSendsAnAuthzExecMessageToUpdateModuleParam(moduleName, paramName string, table gocuke.DataTable) {
+	panic("PENDING")
+}
+
+func (s *suite) TheModuleParamShouldBeUpdated(moduleName, paramName string, table gocuke.DataTable) {
+	panic("PENDING")
+}
+
+func (s *suite) getKeyAddress(keyName string) string {
+	argsAndFlags := []string{
+		"keys",
+		"show",
+		keyName,
+		keyRingFlag,
+		fmt.Sprintf("--%s=json", cometcli.OutputFlag),
+	}
+	res, err := s.pocketd.RunCommandOnHost("local", argsAndFlags...)
+	require.NoError(s, err)
+
+	keyListRes := make(map[string]any)
+	err = json.Unmarshal([]byte(res.Stdout), &keyListRes)
+	require.NoError(s, err)
+
+	return keyListRes["address"].(string)
+}
+
+type anyMap struct {
```

Comment:
NIT: Can you move this out to a helper file. It's not specific to params so we should reuse it elsewhere.